ee/server/service/software_installers.go

```diff
@@ -105,7 +106,13 @@ func (svc *Service) UploadSoftwareInstaller(ctx context.Context, payload *fleet.
 	return nil
 }
 
-func (svc *Service) validateSoftwareLabels(ctx context.Context, labelsIncludeAny, labelsExcludeAny []string) (*fleet.LabelIdentsWithScope, error) {
+func ValidateSoftwareLabels(ctx context.Context, svc fleet.Service, labelsIncludeAny, labelsExcludeAny []string) (*fleet.LabelIdentsWithScope, error) {
```

Comment:
curious: why make this a package function vs a method on `svc`?

Reply:
@jahzielv, This was my way around a package dependency problem I encountered trying to write the unit tests. This function depends on `svc.BatchValidateLabels`, which exists on the core service, which in turn is embedded in the premium service. But for our `ee` tests, I didn't find a good way to actually embed the core service without creating a circular dependency. If you look at a lot of the other tests in `ee`, the [service](https://github.com/fleetdm/fleet/blob/f20bbedf4ff075f1c92407578698ff5633502ecc/ee/server/service/software_installers_test.go#L219-L220) is hollowed out. It essentially boils down to:
```go
	eeservice := &Service{
		Service:           nil,
		ds:                    ds,
		...
	}
```
And the test panics when it tries to call `svc.BatchValidateLabels`. 

I spent a good bit of time looking for a unit-testable approach that would let me keep the software validators in the the `ee` directory as private methods on the premium service, but in the end I opted for this approach. One alternative I found was this [approach](https://github.com/fleetdm/fleet/blob/f20bbedf4ff075f1c92407578698ff5633502ecc/ee/server/service/mdm_export_for_test.go#L14-L15). The downside was that it involved more test [setup](https://github.com/fleetdm/fleet/blob/f20bbedf4ff075f1c92407578698ff5633502ecc/ee/server/service/mdm_external_test.go#L37-L38). Another alternative would be having the methods defined on the core service with license checks as we do with `svc.UploadSoftwareInstaller`, but that felt off to me for helper functions like these. In hindsight, maybe that is fine since I wasn't able to keep them as private methods anyway.